worker/src/main.rs

```diff
@@ -406,3 +446,81 @@ fn worker(node_url: &str, w_ip: &str, w_port: &str, mu_ra_port: &str) {
         }
     }
 }
+
+pub fn process_request(eid: sgx_enclave_id_t, request: Request, node_url: &str) {
+    // new api client (the other one is busy listening to events)
+    // FIXME: this might not be very performant. maybe split into api_listener and api_sender
+    let mut _api = Api::<sr25519::Pair>::new(format!("ws://{}", node_url));
+    info!("*** Ask the signing key from the TEE");
+    let mut signing_key_raw = [0u8; 32];
+    signing_key_raw.copy_from_slice(&enclave_signing_key(eid.clone()).unwrap()[..]);
+
+    // Attention: this HAS to be sr25519, although its a ed25519 key!
```

Comment:
I don't get this, why does this need to be sr25519?

Reply:
because reasons ;-)
....being I couldn't make it work otherwise. I think it has to do with the act that we might actually better define our own AppCrypto but it's not really meant to be used outside of runtime